### ISSUE TITLE
Option for automatically saving track recording when navigation ends

### ISSUE THREAD

**Issue by Zero3** [osmandapp/OsmAnd#12598]
# 🚀 feature request

### Description

I use the "Record tracks automatically during navigation" feature of the track module, which is quite nice for logging all my trips. However, the recording does not ~stop~ save automatically when navigation ends, which I then have to remember to do manually by clicking the small "REC" widget in the top-right part of the screen and click "Finish".

### Describe the solution you'd like

Make an option for the track module that makes it automatically ~stop~ save the track when navigation ends.

### Describe alternatives you've considered

~Stop~ Save when the destination have been reached, instead of all cases of navigation ending. There is a difference in some cases.

A nice bonus feature would be a "Continue anyway" button visible for X seconds after automatically ~stopping~ saving the recording, in the rare case where I want to continue recording anyway.

*(modified by Zero3 to replace "stop" with "save" per discussion below)*

**Comment by vshcherb** [comment]
You mean automatically Save? cause it should stop after navigation has stopped

**Comment by Zero3** [comment]
@vshcherb I am not familiar enough with the plugin to understand the difference between "stopping" and "saving". But yes, I would like for it to automatically stop **and** save after every trip I make.

**Comment by vshcherb** [comment]
Save is a manual action  to create a file, and stop is to stop recording your movement. Save could be done at any moment when you decide that you don't want to continue recording. Automatically save happens on the next day.

**Comment by vshcherb** [comment]
So unless we clarify difference save & stop I'm afraid it's hard to understand if it's a bug of a feature request

**Comment by Zero3** [comment]
@vshcherb

> Save is a manual action to create a file

This is what I would like. So that after every navigation, a new .gpx file is created for just that trip. If I make 3 trips on a day, I thus end up with 3 .gpx files, each containing one of the trips.

**Comment by vshcherb** [comment]
As we understood it's not about ending track, it's about saving and creating file. So this intentionally not implemented so user could continue recording track with new navigation. We could improve situation by waiting 30-40 min and creating track automatically if new navigation didn't start

**Comment by Zero3** [comment]
@vshcherb I see. But won't that add 30-40 minutes of noise at the end of every recorded file? And use battery for GPS for 30-40 minutes as well?

If so, I would still like an option for OsmAnd to automatically stop and save after every navigation. Advanced users who would like multiple navigations in a single .gpx file could turn it off (also see my "continue recording anyway", idea above).

**Comment by vshcherb** [comment]
Nope, recording stops automatically but it doesn't save the file automatically. So there is no noise and GPS usage once navigation is stopped
